service/matching/handler.go

```diff
@@ -298,6 +298,26 @@ func (h *Handler) ListTaskQueuePartitions(
 	return h.engine.ListTaskQueuePartitions(ctx, request)
 }
 
+// UpdateWorkerVersioningRules allows updating the Build ID assignment and redirect rules for a given Task Queue.
+func (h *Handler) UpdateWorkerVersioningRules(
+	ctx context.Context,
+	request *matchingservice.UpdateWorkerVersioningRulesRequest,
+) (_ *matchingservice.UpdateWorkerVersioningRulesResponse, retError error) {
+	defer log.CapturePanic(h.logger, &retError)
+	resp, err := h.engine.UpdateWorkerVersioningRules(ctx, request.GetRequest())
```

Comment:
should we pass the wrapped request so in engine.UpdateWorkerVersioningRules we can make sure the namespace id matches with the resolved id based on the namespace name in the inner request?

is taskqueue needed in the outer request?